src/components/structures/RoomView.tsx

```diff
@@ -306,18 +316,35 @@ export class RoomView extends React.Component<IRoomProps, IRoomState> {
     }
 
     private onWidgetStoreUpdate = () => {
-        if (this.state.room) {
-            this.checkWidgets(this.state.room);
-        }
+        if (!this.state.room) return;
+        this.checkWidgets(this.state.room);
+    };
+
+    private onWidgetEchoStoreUpdate = () => {
+        if (!this.state.room) return;
+        this.checkWidgets(this.state.room);
+    };
+
+    private onWidgetLayoutChange = () => {
+        if (!this.state.room) return;
+        this.checkWidgets(this.state.room); // we cheat here by calling the thing that matters
     };
 
     private checkWidgets = (room) => {
         this.setState({
-            hasPinnedWidgets: WidgetLayoutStore.instance.getContainerWidgets(room, Container.Top).length > 0,
-            showApps: this.shouldShowApps(room),
+            hasPinnedWidgets: WidgetLayoutStore.instance.hasPinnedWidgets(this.state.room),
+            mainSplitContentType: this.getMainSplitContentType(),
+            showApps: this.shouldShowApps(this.state.room),
         });
     };
 
+    private getMainSplitContentType = () => {
+        // TODO-video check if video should be displayed in main panel
```

Comment:
Normally I'd say all these placeholders for a separate project (video) are a bit much (since you never know if project timelines might change), but given what we know at the moment, it does seem helpful, especially since both projects are in the same area of code.

Please make sure to let Robert know you've added these comments once this PR merged. 🙂